src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -3729,6 +3729,10 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           typedFun0
       )
       val treeInfo.Applied(typedFun @ Select(New(annTpt), _), _, _) = typedFunPart
+      if (typedFunPart.isErroneous) {
+        return finish(ErroneousAnnotation)
+      }
```

Comment:
this could more one more line up. and unnecessary `{ }`.

Reply:
It would be terribly stylish to write in one liner:
```
if (typedFunPart.isErroneous) return finish(ErrAnn)
```
Some code bases would frown on one-liners, and maybe also return.

Obviously, "terribly" here means "good".

Reply:
OK, looking at the source, I'd be inclined to avoid return, make the error checks more readable (for instance, only one null check for annType), and instead of the big clever block arg for finish:
```
val info =
  if (check1) Erroneous
  else { other stuff }
finish(info)
```
Maybe it's possible to structure that clearly without early returns?

Not sure you want to undertake that here and now, however.